skimage/filters/_gabor.py

```diff
@@ -13,11 +15,187 @@ def _sigma_prefactor(bandwidth):
+    ----------
+    .. [1] Do CB. 2008. The Multivariate Gaussian Distribution. Stanford
+           University (CS 229): Stanford, CA.
+           http://cs229.stanford.edu/section/gaussians.pdf
+    """
+    sigma_prod = np.prod(sigma)
+
+    # normalization factor
+    norm = (2 * np.pi) ** (ndim / 2)
+    norm *= sigma_prod
+
+    # center image
+    image = image - center
+
+    # gaussian envelope
+    gauss = np.exp(-0.5 * np.sum(image ** 2) / sigma_prod ** 2)
+
+    return gauss / norm
+
+
+def _rotation(src_axis, dst_axis):
+    """Generates an matrix for the nD rotation of one axis to face another.
+
+    Parameters
+    ----------
+    src_axis : (N,) matrix
+        Vector representation of the axis that will be rotated.
+
+    dst_axis : (N,) matrix
+        Vector representation of the axis to rotate to.
+
+    Returns
+    -------
+    M : (N, N) array
+        Matrix that rotates ``src_axis`` to coincide with ``dst_axis``.
+
+    References
+    ----------
+    .. [1] Ognyan Ivanov Zhelezov. One Modification which Increases Performance
+           of N-Dimensional Rotation Matrix Generation Algorithm. International
+           Journal of Chemistry, Mathematics, and Physics, Vol. 2 No. 2, 2018:
+           pp. 13-18. https://dx.doi.org/10.22161/ijcmp.2.2.1
+
+    Examples
+    --------
+    >>> X = np.asarray([1, 0])
+    >>> Y = np.asarray([0.5, 0.5])
+
+    >>> M = _rotation(X, Y)
+    >>> Z = np.matmul(M, Y[..., np.newaxis])
+
+    >>> np.allclose(Z, Y[np.newaxis])
+    True
+    """
+    X = np.array(src_axis)
+    Y = np.array(dst_axis)
+
+
+    def rot(axis, diffs):
+        ndim = len(axis)
+        num_diffs = len(diffs)
+
+        x = axis
+        w = diffs
+
+        R = np.eye(ndim)  # Initial rotation matrix = Identity matrix
+
+        step = 1  # Initial step
+        while step < ndim:  # Loop to create matrices of stages
+            A = np.eye(ndim)
+
+            n = 0
+            while n < ndim - step and n + step < num_diffs:
+                r2 = x[w[n]] * x[w[n]] + x[w[n + step]] * x[w[n + step]]
+                if r2 > 0:
+                    r = np.sqrt(r2)
+                    pcos = x[w[n]] / r  # Calculation of coefficients
+                    psin = -x[w[n + step]] / r
+                    # Base 2-dimensional rotation
+                    A[w[n], w[n]] = pcos
+                    A[w[n], w[n + step]] = -psin
+                    A[w[n + step], w[n]] = psin
+                    A[w[n + step], w[n + step]] = pcos
+                    x[w[n + step]] = 0
+                    x[w[n]] = r
+
+                n += step << 1  # Move to the next base operation
+
+            step <<= 1  # multiply by 2
+            R = np.matmul(A, R)  # Multiply R by current matrix of stage A
+
+        return R
+
+
+    normX = np.linalg.norm(X)
+    normY = np.linalg.norm(Y)
+
+    if not np.isclose(normX, normY):           # Set norm of Y equal to norm
+        Y = (normX / normY) * Y  # of X if they are different
```

Comment:
`Y *= normX / normY`

Reply:
@jni There can be some issues with this shorthand when `Y` is of `dtype=int`:

```python
In [1]: import numpy as np

In [2]: a = np.array([1, 2])

In [3]: a *= 1 / 2
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-831f196b8395> in <module>()
----> 1 a *= 1 / 2

TypeError: Cannot cast ufunc multiply output from dtype('float64') to dtype('int64') with casting rule 'same_kind'

In [4]: a = 1 / 2 * a

In [5]: a
Out[5]: array([ 0.5,  1. ])
```